torch/__init__.py

```diff
@@ -511,24 +511,31 @@ def _sympy_(self):
         return self.node.expr
 
     def __hash__(self) -> builtins.int:
-        return hash(self._get_int())
-
-    def as_integer_ratio(self) -> _Tuple[builtins.int, builtins.int]:
+        if self.node.is_nested_int():
+            return hash(self.node.nested_int())
+        else:
+            # We could support constant SymInts as well, but not doing it for now
+            raise TypeError("unhashable type: non-nested SymInt")
+            # TODO: Force specialization
+            # This can't be done because the TypeError here is load bearing
+            # for einops
+            # https://github.com/arogozhnikov/einops/blob/6181e1e95dc58c00a3143c1726da1c6ee0463164/einops/einops.py#L237
+            # return hash(builtins.int(self))
+
+    def as_integer_ratio(self) -> _Tuple["SymInt", builtins.int]:
         """Represent this int as an exact integer ratio"""
-        return self._get_int(), 1
+        return self, 1
 
-    def bit_length(self) -> "SymInt":
-        return SymInt(self.node.wrap_int(self._get_int().bit_length()))
+    def bit_length(self) -> builtins.int:
+        # TODO: A more relaxed guard is possible here, where you guard to
+        # allow all integer quantities which would result in the same bit
+        # length.  We can also just make a dedicated Sympy function for
+        # computing this quantity and represent it symbolically.
+        return builtins.int(self).bit_length()
```

Comment:
oh, I thought that the bug was coming from somewhere else. I guess this should be
```suggestion
        v = self.node.nested_int() if self.node.is_nested_int() else builtins.int(self)
        return v.bit_length()
```
which is what `_get_int` was doing

Reply:
I'm a little unclear if anything is now expected of me. :-)

Just ping me on this thread, or file an issue with my name on it, if further action is needed.

Reply:
Nothing from you for now :)